tools-local/tasks/ProcessSharedFrameworkDeps.cs

```diff
@@ -0,0 +1,68 @@
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Extensions.DependencyModel;
+using NuGet.Common;
+using NuGet.ProjectModel;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class ProcessSharedFrameworkDeps : Task
+    {
+        [Required]
+        public string AssetsFilePath { get; set; }
+
+        [Required]
+        public string DepsFilePath { get; set; }
+
+        [Required]
+        public ITaskItem[] PackagesToRemove { get; set; }
+
+        [Required]
+        public ITaskItem[] Runtimes { get; set; }
+
+        public override bool Execute()
+        {
+            DependencyContext context;
+            using (var depsStream = File.OpenRead(DepsFilePath))
+            {
+                context = new DependencyContextJsonReader().Read(depsStream);
+            }
+            LockFile lockFile = LockFileUtilities.GetLockFile(AssetsFilePath, NullLogger.Instance);
+
+            var manager = new RuntimeGraphManager();
+            var graph = manager.Collect(lockFile);
+            var expandedGraph = manager.Expand(graph, ToStringArray(Runtimes));
+
+            var trimmedRuntimeLibraries = context.RuntimeLibraries;
+
+            if (PackagesToRemove != null && PackagesToRemove.Any())
+            {
+                trimmedRuntimeLibraries = RuntimeReference.RemoveReferences(context.RuntimeLibraries, ToStringArray(PackagesToRemove));
+            }
+
+            context = new DependencyContext(
+                context.Target,
+                context.CompilationOptions,
+                context.CompileLibraries,
+                trimmedRuntimeLibraries,
+                expandedGraph
+                );
+
+            using (var depsStream = File.Create(DepsFilePath))
+            {
+                new DependencyContextWriter().Write(context, depsStream);
+            }
+
+            return true;
+        }
+
+        private static string[] ToStringArray(ITaskItem[] items)
```

Comment:
I think you can just change the types of the parameters to the task to `string[]` instead of `ITaskItem[]` and MSBuild will do the conversion for you.  Doesn't look like there's a need to be able to access any Metadata on the items, so this should be safe.